src/rdlist.c

```diff
@@ -246,6 +246,10 @@ int rd_list_cmp_trampoline (const void *_a, const void *_b) {
 }
 
 void rd_list_sort (rd_list_t *rl, int (*cmp) (const void *, const void *)) {
+	if (rl->rl_elems == NULL) {
```

Comment:
I know it is a mix of tabs and spaces in the code, and trhat will be fixed, but new code should use 8 whitespaces for indent.
Also drop the curly braces, we don't use them for single line blocks.

Reply:
and add unlikely()

Reply:
you should look in to clang-format, it's pretty good tool for this

Reply:
I did last year, but I couldn't get it to completely match the emacs c-style linux.